### ISSUE TITLE
33512114: -setAccessibilityFocus: on text elements and disabled buttons has virtually no effect

### ISSUE THREAD

**Issue by openradar-mirror** [lionheart/openradar-mirror#18030]
#### Description

Area:
Accessibility

Summary:
Programmatically moving the accessibility focus to accessibility elements does not work if the elements are text fields or dimmed buttons, although the property has been successfully set to YES.

Steps to Reproduce:
- Open the attached project.
- Start VoiceOver.
- Run the project.
- In the console you will see which element has been programmatically selected.

Expected Results:
VoiceOver should cycle through all three elements (the disabled prevViewButton, the enabled nextViewButton and the multilineLabel), three times with one second between each switch, as specified in the code.

Observed Results:
VoiceOver only selects the enabled nextViewButton.

Notes:
The only way of programmatically selecting the multilineLabel is by setting the accessibilityElement property of both buttons to NO, so the label is the only available element.

Being able to select the label programmatically is important, because labels can contain important information, while the buttons may just say "Agree" or "Cancel", which without context only confuses the user.

The attached file can be downloaded here:
https://peporal.de/bugs/Accessibility%20Focus.zip

-
Product Version: macOS 10.13 Beta (17A315i)
Created: 2017-07-25T14:42:34.046400
Originated: 2017-07-25T00:00:00
Open Radar Link: http://www.openradar.me/33512114
